iceoryx_utils/include/iceoryx_utils/internal/cxx/storable_function.inl

```diff
@@ -0,0 +1,264 @@
+        // note: vtable is needed for destroy, then changed to src vtable
+        m_vtable.destroy(*this);
+        m_function = nullptr; // only needed when the src has no object
+        m_vtable = rhs.m_vtable;
+        m_vtable.copy(rhs, *this);
+    }
+
+    return *this;
+}
+
+template <typename S, typename ReturnType, typename... Args>
+storable_function<S, signature<ReturnType, Args...>>&
+storable_function<S, signature<ReturnType, Args...>>::operator=(storable_function&& rhs) noexcept
+{
+    if (&rhs != this)
+    {
+        // note: dest (this) vtable is needed for destroy, then changed to source (rhs) vtable
+        m_vtable.destroy(*this);
+        m_function = nullptr; // only needed when the source has no object
+        m_vtable = rhs.m_vtable;
+        m_vtable.move(rhs, *this);
+    }
+
+    return *this;
+}
+
+template <typename S, typename ReturnType, typename... Args>
+storable_function<S, signature<ReturnType, Args...>>::~storable_function() noexcept
+{
+    m_vtable.destroy(*this);
+}
+
+
+template <typename S, typename ReturnType, typename... Args>
+ReturnType storable_function<S, signature<ReturnType, Args...>>::operator()(Args... args)
+{
+    auto r = m_function(std::forward<Args>(args)...);
+    return r;
+}
+
+template <typename S, typename ReturnType, typename... Args>
+storable_function<S, signature<ReturnType, Args...>>::operator bool() noexcept
+{
+    return m_function.operator bool();
+}
+
+template <typename S, typename ReturnType, typename... Args>
+void storable_function<S, signature<ReturnType, Args...>>::swap(storable_function& f) noexcept
+{
+    storable_function tmp = std::move(f);
+    f = std::move(*this);
+    *this = std::move(tmp);
+}
+
+template <typename S, typename ReturnType, typename... Args>
+void storable_function<S, signature<ReturnType, Args...>>::swap(storable_function& f, storable_function& g) noexcept
+{
+    storable_function tmp = std::move(f);
+    f = std::move(g);
+    g = std::move(tmp);
+}
+
+template <typename S, typename ReturnType, typename... Args>
+template <typename Functor, typename>
+void storable_function<S, signature<ReturnType, Args...>>::storeFunctor(const Functor& functor) noexcept
+{
+    ///@todo we may need something more to get the correct type in general...
+    using StoredType = typename std::remove_reference<Functor>::type;
+    auto p = m_storage.template allocate<StoredType>();
+
+    if (p)
+    {
+        // functor will fit, copy it
+        p = new (p) StoredType(functor);
+
+        // erase the functor type and store as reference to the call in storage
+        m_function = *p;
+        m_storedObj = p;
+        m_vtable.copyFunction = copy<StoredType>;
+        m_vtable.moveFunction = move<StoredType>;
+        m_vtable.destroyFunction = destroy<StoredType>;
+    }
+
+    // else we detect the problem at compile time or store nothing when memory is exhausted
+    // note that we have no other choice, it is used in the ctor and we cannot throw
```

Comment:
Can do. As noted below, for the usage with static_storage (as defined in function.hpp which is the version we will be using) this will never be a problem (it will detect that there is not enough storage at compile time and fail to compile).

Of course, there always is the possibility of implementation error but I tried locally to use it with callables that are too large and it fails to compile (we cannot test this properly with unit tests).

Reply:
Deleted comment, added note to the function and ctor describing the behavior. Note tha in the copy ctor/assignment this must will always work with static_storage (since it worked for the original).